ibm/service/cos/resource_ibm_cos_bucket_object.go

```diff
@@ -269,30 +318,61 @@ func resourceIBMCOSBucketObjectRead(ctx context.Context, d *schema.ResourceData,
 		log.Printf("[INFO] Ignoring body of COS bucket (%s) object (%s) with Content-Type %q", bucketName, objectKey, contentType)
 	}
 
+	getObjectRetentionInput := new(s3.GetObjectRetentionInput)
+	getObjectRetentionInput.Bucket = aws.String(bucketName)
+	getObjectRetentionInput.Key = aws.String(objectKey)
+	response, err := s3Client.GetObjectRetention(getObjectRetentionInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "NotFound" {
```

Comment:
if retention is not able to find we shou;dn't set ID to empty because object still exists right?

Reply:
Yes , the object will still exist.

Reply:
Addressed the comment.